pxf/pxf-service/src/main/java/org/apache/hawq/pxf/service/utilities/ProtocolData.java

```diff
@@ -82,19 +83,26 @@ public ProtocolData(Map<String, String> paramsMap) {
         parseTupleDescription();
 
         /*
-         * accessor - will throw exception from getPropery() if outputFormat is
+         * accessor - will throw exception if outputFormat is
          * BINARY and the user did not supply accessor=... or profile=...
-         * resolver - will throw exception from getPropery() if outputFormat is
+         * resolver - will throw exception if outputFormat is
          * BINARY and the user did not supply resolver=... or profile=...
          */
-        profile = getOptionalProperty("PROFILE");
+        profile = getUserProperty("PROFILE");
         if (profile != null) {
             setProfilePlugins();
         }
-        accessor = getProperty("ACCESSOR");
-        resolver = getProperty("RESOLVER");
-        fragmenter = getOptionalProperty("FRAGMENTER");
-        metadata = getOptionalProperty("METADATA");
+        accessor = getUserProperty("ACCESSOR");
+        if(accessor == null) {
+            protocolViolation(accessor);
```

Comment:
what's the point of sending null to protocolViolation function ? would you want to send string instead ?